src/Validation.resi

```diff
@@ -0,0 +1,3 @@
+type rec t
+let syncFunction: (string => bool) => t
+let asyncFunction: (string => Js.Promise.t<bool>) => t
```

Comment:
I like this version better than the union as it comes with no runtime cost 😄 
Just a nitpick: would you mind calling these sync/async? (and maybe add a line return in between the different lines of code 🙏 )

Reply:
Thank you done